contracts/vaults/GammaVault/RibbonGammaVault.sol

```diff
@@ -0,0 +1,996 @@
+            );
+
+            vaultState.totalPending = 0;
+            vaultState.round = uint16(currentRound + 1);
+        }
+
+        _mint(address(this), mintShares);
+
+        if (totalVaultFee > 0) {
+            IERC20(USDC).safeTransfer(recipient, totalVaultFee);
+        }
+
+        lastQueuedWithdrawAmount = queuedWithdrawAmount;
+
+        ShareMath.assertUint104(lockedBalance);
+        vaultState.lockedAmount = uint104(lockedBalance);
+
+        newRoundInProgress = true;
+    }
+
+    /**
+     * @notice Swaps pending USDC deposits into WETH
+     * @param amountIn Amount of USDC to swap into WETH
+     * @param minAmountOut Minimum amount of WETH to receive
+     * @return amountOut Amount of WETH received from the swap
+     */
+    function swapTotalPending(uint256 amountIn, uint256 minAmountOut)
+        external
+        onlyKeeper
+        nonReentrant
+        returns (uint256 amountOut)
+    {
+        require(newRoundInProgress, "!newRoundInProgress");
+        require(amountIn > 0, "!amountIn");
+        require(minAmountOut > 0, "!minAmountOut");
+
+        // Deduct queued withdrawals from the balance
+        uint256 _totalPending = vaultState.totalPending;
+        if (amountIn > _totalPending) amountIn = _totalPending;
+
+        // Swap pending USDC deposits to WETH
+        amountOut = UniswapRouter.swap(
+            address(this),
+            USDC,
+            amountIn,
+            minAmountOut,
+            UNISWAP_ROUTER,
+            usdcWethSwapPath
+        );
+
+        _totalPending = _totalPending.sub(amountIn);
+        ShareMath.assertUint128(_totalPending);
+
+        vaultState.totalPending = uint128(_totalPending);
+    }
+
+    /**
+     * @notice Deposit WETH into the squeeth vault as collateral to mint SQTH
+     * @param wethAmount Amount of WETH to deposit
+     * @param minAmountOut Minimum amount of WETH to receive from swapping minted SQTH to WETH
+     */
+    function depositTotalPending(uint256 wethAmount, uint256 minAmountOut)
+        external
+        onlyKeeper
+        nonReentrant
+    {
+        require(newRoundInProgress, "!newRoundInProgress");
+        require(wethAmount > 0, "!wethAmount");
+        require(minAmountOut > 0, "!minAmountOut");
+
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (wethAmount > wethBalance) wethAmount = wethBalance;
+
+        uint256 wethUsdcPrice =
+            VaultLifecycleGamma.getWethPrice(
+                ORACLE,
+                USDC_WETH_POOL,
+                WETH,
+                USDC
+            );
+        uint256 sqthAmount =
+            VaultLifecycleGamma.getSqthMintAmount(
+                CONTROLLER,
+                wethUsdcPrice,
+                COLLATERAL_RATIO,
+                wethAmount
+            );
+
+        // Deposit ETH collateral and mint oSQTH
+        IWETH(WETH).withdraw(wethAmount);
+        IController(CONTROLLER).mintWPowerPerpAmount{value: wethAmount}(
```

Comment:
Could you not call `mintWPowerPerpAmount` with an excessive wethAmount? Rather than doing a separate `deposit` call to top up the collateral necessary